### PR TITLE
Python Implementation of some algorithms

### PR DESCRIPTION
**Pull request by gaurav-12**
Adding python implementation of some algorithms (BFS, Heap Sort, Selection Sort, Insertion Sort and more).

### CODE REVIEW

**Comment by gaurav-12** [comment]
Hi @starkblaze01 ,
I have removed, Merge, Insertion, Selection Sorts and BFS in the [latest commit](https://github.com/starkblaze01/Algorithms/pull/83/commits/28c33e51f5dd5d91047c42b9fe27dfb8156ef098)

File: HeapSort_Py/HeapSort.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+        child = arr[i - 1]
+        
+        print('Parent of ' + str(child) + ' : ' + str(parent))
+
+        if(parent < child):
+            arr[i//2 - 1], arr[i - 1] = child, parent
+    return arr
+
+arr = [20,10,6,2,5,3,15]
+print(HeapSort(arr))
```
Comment (starkblaze01) [inline HeapSort_Py/HeapSort.py]: Are you sure this algorithm is complete?
Because the output array is not sorted.

Comment (gaurav-12) [reply HeapSort_Py/HeapSort.py]: Yes, the algorithm is complete, it gives a tree as an output in the form of Array.
Considering the arr given in program, 20 is the root, with two children, 10 and 6, and 10 have two children, 2 and 5, and so on. 
Though it is limited to parents with only two children, it does sort the tree, where each parent's left is smaller than itself and the right will be bigger.

Comment (starkblaze01) [reply HeapSort_Py/HeapSort.py]: The output for your example is [20, 10, 15, 2, 5, 3, 6].
If we want to form the tree then 20 will be the root node and 10 will be its left child and 15 will be right. But 15 is smaller than 20 which should be greater if it's sorted.

Comment (gaurav-12) [reply HeapSort_Py/HeapSort.py]: Yes, you are right, it is not working as expected.
Thanks for the review, will remove it and add it later when complete.

**Pull request merged**